pkg/costattribution/active_tracker.go

```diff
@@ -110,24 +147,47 @@ func (at *ActiveSeriesTracker) Increment(lbls labels.Labels, now time.Time) {
 
 	as, ok = at.observed[string(buf.Bytes())]
 	if ok {
-		as.Inc()
+		as.asCount.Inc()
+		if nativeHistogramBucketNum >= 0 {
+			as.nhCount.Inc()
+			as.nhBucketNum.Add(int64(nativeHistogramBucketNum))
+		}
 		return
 	}
 
 	if !at.overflowSince.IsZero() {
-		at.overflowCounter.Inc()
+		at.overflowCounter.asCount.Inc()
+		if nativeHistogramBucketNum >= 0 {
+			at.overflowCounter.nhCount.Inc()
+			at.overflowCounter.nhBucketNum.Add(int64(nativeHistogramBucketNum))
+		}
 		return
 	}
 
 	if len(at.observed) >= at.maxCardinality {
 		at.overflowSince = now
-		at.overflowCounter.Inc()
+		at.overflowCounter.asCount.Inc()
+		if nativeHistogramBucketNum >= 0 {
+			at.overflowCounter.nhCount.Inc()
+			at.overflowCounter.nhBucketNum.Add(int64(nativeHistogramBucketNum))
+		}
 		return
 	}
-	at.observed[string(buf.Bytes())] = atomic.NewInt64(1)
+
+	nhCount := atomic.NewInt64(0)
+	nhBucketNum := atomic.NewInt64(0)
+	if nativeHistogramBucketNum >= 0 {
+		nhCount.Inc()
+		nhBucketNum.Add(int64(nativeHistogramBucketNum))
+	}
+	at.observed[string(buf.Bytes())] = asMetrics{
+		asCount:     atomic.NewInt64(1),
+		nhCount:     nhCount,
+		nhBucketNum: nhBucketNum,
+	}
```

Comment:
nit, this looks a bit faster:
```suggestion

	nhCount := int64(0)
	nhBucketNum := int64(0)
	if nativeHistogramBucketNum >= 0 {
		nhCount++
		nhBucketNum+=int64(nativeHistogramBucketNum)
	}
	at.observed[string(buf.Bytes())] = asMetrics{
		asCount:     atomic.NewInt64(1),
		nhCount:     atomic.NewInt64(nhCount),
		nhBucketNum: atomic.NewInt64(nhBucketNum),
	}
```